pkg/kubelet/cri/remote/utils_test.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package remote
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	runtimeapi "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
+)
+
+func TestVerifySandboxStatus(t *testing.T) {
+	validStatus := getValidPodSandboxStatus()
+	statusWithoutID := getValidPodSandboxStatus()
+	statusWithoutID.Id = ""
+	statusWithoutMetadata := getValidPodSandboxStatus()
+	statusWithoutMetadata.Metadata = nil
+	statusWithoutCreatedAt := getValidPodSandboxStatus()
+	statusWithoutCreatedAt.CreatedAt = 0
+	statusWithoutName := getValidPodSandboxStatus()
+	statusWithoutName.Metadata.Name = ""
```

Comment:
Is it possible to just inline these? I found these tests a little difficult to read since I had to read in two different places for the context in each test case.

Reply:
I could not follow what I should do here.
Do you have any sample unit test code in the other place?

Reply:
I'm not sure what the idiomatic way to do this in Go is. I looked at a few other unit tests but didn't find anything that would help here. I suppose you could change the podSandboxStatus to be a function instead that you call in the test body, and then you could write anonymous functions for these.

Reply:
@oomichi chatted with someone a bit more experienced with Go. My suggestion would be to update `getValidPodSandboxStatus` (and its counterpart `getValidContainerStatus`) to take arguments so then you can simply call the function with arguments in each test case below rather than having to do this.

Reply:
I don't think that is better than current one because we need to specify all argument values which need to be changed for all test cases.
That doesn't make the code readable.

Reply:
Right now, this is not particularly readable because the test case parameters aren't actually defined in each test case.

Reply:
Do you think it is worth to put `/hold` for the above point?

Reply:
I think the above point is applied on the latest PR.

/hold cancel